lib/src/client/connection.dart

```diff
@@ -13,11 +14,70 @@ class Connection {
   /// The stream socket subscription.
   final StreamSubscription<List<int>> _subscription;
 
+  /// Error handler from the latest listener.
+  void Function(Object, StackTrace) _onErrorListener;
+
+  /// Future that is resolved when the connection is closed.
+  ///
+  /// If the connection closes gracefully this future will be completed and
+  /// future attempts to use the connection will throw a
+  /// [ConnectionClosedException].
+  ///
+  /// If the connection is broken, an error occurs writing/reading to/from the
+  /// connection this future will be resolved with error. If unhandled this
+  /// will propogate to the encapsulating [Zone] where it may be handled.
+  /// If using the default root [Zone] this will cause the isolate to crash.
+  /// If not implementing custom reconnection logic it might be desirable to
+  /// simply restart the process when it crashes.
+  ///
+  /// If implementing custom reconnection logic, consumers should stop using a
+  /// connection once [done] have been resolved, as all future commands will
+  /// will throw an error. If an error occurs reading or writing all outstanding
+  /// commands will be resolved with an exception.
+  Future<void> get done => _done.future;
+
+  /// Implementation of done.
+  final Completer<void> _done = Completer<void>();
+
   /// Creates a [Connection] instance with the given socket.
   ///
+  /// The [onError] handler is called when a permanent error occurs, from then
```

Comment:
```suggestion
  /// The [_onError] handler is called when a permanent error occurs, from then
```
Travis is crashing due that reference.

Reply:
Uh, my bad.. I meant to remove that paragraph, it was from a previous iteration where I allowed the user to provide an error handler in the constructor..

But I moved away from that design and added the `done` property instead, which if unawaited will crash the process...

I'll push an update.. thanks for the quick turnaround :)